### PR TITLE
doctest: make the "Exporting a C Library" example work

### PR DESCRIPTION
**Pull request by perillo**
The mathtest example of exporting a library with the C ABI it not working, since the mathtest.h file is not generated.

Update the shell sample, adding the -femit-h flag to the zig build-lib command.

Show an example of how to copy the zig.h file to the project directory, using zig env and jq.  I'm not sure if there is a better/official method.

### CODE REVIEW

File: doc/langref.html.in [hunk 1]
```diff
@@ -10901,9 +10901,15 @@ export fn add(a: i32, b: i32) i32 {
...
+      {#shell_samp#}$ zig build-lib -femit-h mathtest.zig{#end_shell_samp#}
       <p>To make a shared library:</p>
-      {#shell_samp#}$ zig build-lib mathtest.zig -dynamic{#end_shell_samp#}
+      {#shell_samp#}$ zig build-lib -femit-h mathtest.zig -dynamic{#end_shell_samp#}
+      <p>
+      The <kbd>-femit-h</kbd> flag is necessary in order to generate the mathtest.h file.
+      The generated C header file includes the zig.h file that must be copied
+      to the project directory, as an example using <a href="https://stedolan.github.io/jq/">jq</a>:
+      </p>
+      {#shell_samp#}cp $(zig env | jq -r .lib_dir)/zig.h .{#end_shell_samp#}
```
Comment (Vexu) [inline doc/langref.html.in:10912]: A command like this shouldn't be needed to use `-femit-h` and should be fixed before documenting `-femit-h`.

**Review by Vexu — CHANGES_REQUESTED** [review]

File: doc/langref.html.in (hunk 1 above)
Comment (perillo) [reply doc/langref.html.in:10912]: The command is not needed to use `-femit-h`, but to copy the `zig.h` file to the current directory.  Without the `zig.h` file `zig build` fails on my system.

Thanks.

Comment (perillo) [reply doc/langref.html.in:10912]: Did you set the `CPATH` environment variable manually?

I also tried with the new official zig 0.10.0 package from Archlinux, but `zig.h` is not found (it is installed in `/usr/lib/zig/include/zig.h`.

Comment (Vexu) [reply doc/langref.html.in:10912]: I didn't set anything, I get all the same errors as you do. What I meant is that you shouldn't need to copy anything or set some environment variable, `zig.h` should either not be referenced by the emitted header or it should be installed alongside it.

Comment (perillo) [reply doc/langref.html.in:10912]: Ok, I misunderstood what you wrote.  Thanks.
I will open an issue.

**Comment by perillo** [comment]
I found an open issue for the same problem of `mathtest.h` not being generated: #6753.

**Comment by perillo** [comment]
FYI: In addition to the the "Exporting a C Library" example, there is another not working example in the "Mixing Object Files" example.

**Pull request closed without merging**